docs/llm_tools_invocation.md

```diff
@@ -0,0 +1,114 @@
+| `context`         | Context | The `homeassistant.core.Context` of the conversation                                                    |
+| `user_prompt`     | string  | The raw text input that initiated the tool call                                                         |
+| `language`        | string  | The language of the conversation agent, or "*" for any language                                         |
+| `assistant`       | string  | The assistant name used to control exposed entities. Currently only `conversation` is supported.        |
+
+## Getting available tools
+
+You can get the currently registered tools with `homeassistant.helpers.llm.async_get_tools(hass)`.
+
+If need the tool specification in OpenAPI-compatible format, you can use `voluptuous_openapi.convert` function (don't forget to mention the dependency in your `manifest.json` file).
+
+Example:
+```python
+from voluptuous_openapi import convert
+from homeassistant.helpers import llm
+...
+        tools = [
+            {
+                "name": tool.name,
+                "description": tool.description,
+                "parameters": convert(tool.parameters),
+            }
+            for tool in llm.async_get_tools(self.hass)
+        ]
+```
+
+Refer to the LLM API specification for the correct format.
+
+## Calling the tool
+
+When the LLM response indicates it wants to call a tool, you need to prepare the `llm.ToolInput` object, then execute the `llm.async_call_tool` function.
+
+Example:
+
+```python
+from homeassistant.helpers import llm
+
+class MyConversationEntity(
+    conversation.ConversationEntity, conversation.AbstractConversationAgent
+):
+    """Represent a conversation entity."""
+
+    async def async_process(
+        self, user_input: conversation.ConversationInput
+    ) -> conversation.ConversationResult:
+        """Process a sentence."""
+
+    while True:
+        response = ... # Get response from your LLM
+        if not response.tool_calls:
+            break
+
+        for tool_call in response.tool_calls:
+            LOGGER.info(
+                "Tool call: %s(%s)",
+                tool_call.function.name,
+                tool_call.function.arguments,
+            )
+            tool_input = llm.ToolInput(
+                tool_name=tool_call.function.name,
+                tool_args=json.loads(tool_call.function.arguments),
+                platform=DOMAIN,
+                context=user_input.context,
+                user_prompt=user_input.text,
+                language=user_input.language,
+                assistant=conversation.DOMAIN,
+            )
+            try:
+                tool_response = await llm.async_call_tool(
+                    self.hass, tool_input
+                )
+            except (HomeAssistantError, vol.Invalid) as e:
+                tool_response = {"error": type(e).__name__}
```

Comment:
Make a `ConversationResult` here?

Reply:
Too early. Need to feed the tool result back to LLM and wait for its response.